src/backend/distributed/commands/multi_copy.c

```diff
@@ -2330,6 +2396,30 @@ CitusSendTupleToPlacements(TupleTableSlot *slot, CitusCopyDestReceiver *copyDest
 }
 
 
+/*
+ * ContainsLocalPlacement returns true if the current node has
+ * a local placement for the given shard id.
+ */
+static bool
+ContainsLocalPlacement(int64 shardId)
+{
+	ListCell *placementCell = NULL;
+	List *activePlacementList = ActiveShardPlacementList(shardId);
+	int32 localGroupId = GetLocalGroupId();
+
+	foreach(placementCell, activePlacementList)
```

Comment:
We have started to prefer `foreach_ptr` to `foreach`. I think `foreach_ptr` wasn't available in postgres few releases back, so you still see `foreach` in Citus code. But now postgres 11 and 12 both has `foreach_ptr`.

Reply:
I would suggest that we change this with another PR because it is not introduced with this PR and just changing this one seems a bit weird to me